tokio/src/task/mod.rs

```diff
@@ -69,11 +74,18 @@ pub(crate) trait Schedule: Send + Sync + Sized + 'static {
     fn schedule(&self, task: Task<Self>);
 }
 
+/// Marker trait indicating that a scheduler can only schedule tasks which
+/// implement `Send`.
+///
+/// Schedulers that implement this trait may not schedule `!Send` futures. If
+/// trait is implemented, the corresponding `Task` type will implement `Send`.
+pub(crate) trait ScheduleSend: Schedule + Send + Sync {}
+
 /// Create a new task without an associated join handle
 pub(crate) fn background<T, S>(task: T) -> Task<S>
 where
     T: Future + Send + 'static,
-    S: Schedule,
+    S: Schedule + ScheduleSend,
```

Comment:
I think `Schedule +` is redundant here (and elsewhere) given the definition for `ScheduleSend`.

Reply:
yup, changed it. although, IMO, this seems like it makes more sense if we leave the trait `ScheduleSend` (rather than `RequiresSend` as discussed in https://github.com/tokio-rs/tokio/pull/1733#discussion_r346072005); a `S: RequiresSend` bound doesn't really tell me that `S` is a scheduler, IMO...

Reply:
alternatively: `ScheduleSendOnly`? It is a little wordier, but implies that it can _only_ schedule `Send` tasks, without removing the word `Schedule`, which I think is important if we remove the `Schedule` bound here?

Reply:
Yeah, that seems a little better to me at least.